src/edu/psu/sweng500/team8/play/ActionManager.java

```diff
@@ -11,20 +11,20 @@
 	 * so they can be adjusted on the fly. 
 	 * Or we can still include it in the JAR but as a separate file
 	 * so we can change it through the development process.  */
-	private static final int QUEUE_SIZE_LIMIT = 10;
+	private static final int STACK_SIZE_LIMIT = 10;
 	
-	private Stack<SudokuAction> sudokuActionQueue;
-	private Stack<SudokuAction> sudokuActionQueueForUndo;
+	private Stack<SudokuAction> sudokuActionStack;
+	private Stack<SudokuAction> sudokuActionStackForUndo;
 
 	public ActionManager() {
-		this.sudokuActionQueue = new Stack<SudokuAction>();
-		this.sudokuActionQueue.setSize(QUEUE_SIZE_LIMIT);		
-		this.sudokuActionQueueForUndo = new Stack<SudokuAction>();
-		this.sudokuActionQueueForUndo.setSize(QUEUE_SIZE_LIMIT);
+		this.sudokuActionStack = new Stack<SudokuAction>();
+		this.sudokuActionStack.setSize(STACK_SIZE_LIMIT);		

```

Comment:
We could use ArrayDeque, except you had a concern of undo/redo stack
growing without bound before. Just like ArrayList  vs. Array ([]), when
there is a need to restrict the size, we need to pick one over the other.
And if we don't worry about size restriction isEmpty works just as well
with Satck.
I know Stack is a pretty bad fruit in the Java world and I know all the
(supposedly) benefits of ArrayDeque. But none of them is critical enough to
change the code after coded/tested.
And depending on to whom you talk, ArrayDeque is not a good pick because it
holds two responsibilities (LIFO & FIFO) which then again could potentially
confuse future Dev.
On Jul 7, 2015 7:50 AM, "Jeremy Nelson" <notifications@github.com> wrote:

> In src/edu/psu/sweng500/team8/play/ActionManager.java
> <https://github.com/garnryang/Team8/pull/10#discussion_r34031351>:
>
> >
> >  	public ActionManager() {
> > -		this.sudokuActionQueue = new Stack<SudokuAction>();
> > -		this.sudokuActionQueue.setSize(QUEUE_SIZE_LIMIT);		
> > -		this.sudokuActionQueueForUndo = new Stack<SudokuAction>();
> > -		this.sudokuActionQueueForUndo.setSize(QUEUE_SIZE_LIMIT);
> > +		this.sudokuActionStack = new Stack<SudokuAction>();
> > +		this.sudokuActionStack.setSize(STACK_SIZE_LIMIT);		
>
> The java standard recommends using ArrayDeque instead of Stack. Then you
> don't need to worry about the initial size (because resizing is fast), and
> isEmpty will work for you.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/garnryang/Team8/pull/10/files#r34031351>.
>